### PR TITLE
Add GroupVersionKind extension to OpenAPI operations

### PR DESCRIPTION
**Pull request by mbohlool**
Fixes: #43249
ref: #34254

```release-note
Added Group/Version/Kind and Action extension to OpenAPI Operations 
```

### CODE REVIEW

**Comment by pwittrock** [comment]
Woot!  This is awesome.

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 1]
```diff
@@ -43,7 +43,8 @@ import (
 )
 
 const (
-	ROUTE_META_GVK = "x-kuberentes-group-version-kind"
+	ROUTE_META_GVK    = "x-kuberentes-group-version-kind"
+	ROUTE_META_ACTION = "x-kuberentes-action"
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: s/kuberentes/kubernetes/g?

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go [hunk 2]
```diff
@@ -800,6 +801,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		}
 		for _, route := range routes {
 			route.Metadata(ROUTE_META_GVK, reqScope.Kind)
+			route.Metadata(ROUTE_META_ACTION, action.Verb)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: seems like this should match the verbs used by authorization and published in discovery

Comment (mbohlool) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: Not sure what you mean, where can I find those verbs?

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go (hunk 1 above)
Comment (mbohlool) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: Done.

**Comment by pwittrock** [comment]
Expected test failures?

**Comment by mbohlool** [comment]
It constantly need rebase :(

**Comment by mbohlool** [comment]
@lavalamp This PR is ready for review (tests are finally green after submitting godep update separately).

File: staging/src/k8s.io/apiserver/pkg/endpoints/installer.go (hunk 2 above)
Comment (mbohlool) [inline staging/src/k8s.io/apiserver/pkg/endpoints/installer.go]: @liggitt friendly ping. Can you explain what you mean here?

File: api/openapi-spec/swagger.json [hunk 3]
```diff
@@ -226,6 +238,12 @@
       "401": {
        "description": "Unauthorized"
       }
+     },
+     "x-kubernetes-action": "LIST",
+     "x-kubernetes-group-version-kind": {
+      "Group": "",
+      "Version": "v1",
+      "Kind": "ConfigMap"
```
Comment (lavalamp) [inline api/openapi-spec/swagger.json]: Shouldn't this be ConfigMapList?

**Comment by lavalamp** [comment]
Just the one comment. Might not be a bad idea to also state the kind of the elements of lists?

File: api/openapi-spec/swagger.json (hunk 3 above)
Comment (pwittrock) [inline api/openapi-spec/swagger.json]: The kind is not the response type, it is the resource Kind that this operation is for.  So I think the Kind is correct here (operation List, resource ConfigMap).  The response type is included in the non-extension portion of the API operation.

**Comment by pwittrock** [comment]
@lavalamp The List kind is present in the operation response kind.  Is this sufficient, or were you linking of some other semantic??

**Comment by mbohlool** [comment]
I agree with @pwittrock, `gvk` extension is not body or response type (those are part of normal spec), they are the API object this operation operating on. These extensions are good for create category of operations on API Objects (e.g. Pod operations may accept or return Pod, PodList, etc.)

**Review by pwittrock — APPROVED** [review]

**Comment by pwittrock** [comment]
Not sure what this new GH approval feature is.  LGTM, will defer to Daniel for final approval.

**Pull request merged**